app/src/main/java/com/arctouch/codechallenge/home/HomeViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.arctouch.codechallenge.home
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.arctouch.codechallenge.AppApplication.Companion.api
+import com.arctouch.codechallenge.util.Constants.Companion.API_KEY
+import com.arctouch.codechallenge.util.Constants.Companion.DEFAULT_LANGUAGE
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.model.TopRatedMoviesResponse
+import com.arctouch.codechallenge.util.Constants.Companion.DEFAULT_REGION
+import com.arctouch.codechallenge.util.Constants.Companion.PAGE_START
+
+class HomeViewModel : ViewModel(){
```

Comment:
Just check my previous comment on regarding how to use Repository pattern here to abstract networking related code outside the `Application` class.

Reply:
Ok, thank you!